contracts/core/IAllo.sol

```diff
@@ -45,55 +64,180 @@ interface IAllo {
         uint256 amount,
         Metadata metadata
     );
+
+    // @dev Event emitted when a pools metadata is updated
     event PoolMetadataUpdated(uint256 indexed poolId, Metadata metadata);
+
+    // @dev Event emitted when a pool is funded
     event PoolFunded(uint256 indexed poolId, uint256 amount, uint256 fee);
+
+    // @dev Event emitted when the base fee is paid
     event BaseFeePaid(uint256 indexed poolId, uint256 amount);
+
+    // @dev Event emitted when the treasury address is updated
     event TreasuryUpdated(address treasury);
+
+    // @dev Event emitted when the percent fee is updated
     event PercentFeeUpdated(uint256 percentFee);
+
+    // @dev Event emitted when the base fee is updated
     event BaseFeeUpdated(uint256 baseFee);
+
+    // @dev Event emitted when the registry address is updated
     event RegistryUpdated(address registry);
+
+    // @dev Event emitted when a strategy is approved and added to the cloneable strategies mapping
     event StrategyApproved(address strategy);
+
+    // @dev Event emitted when a strategy is removed from the cloneable strategies mapping
     event StrategyRemoved(address strategy);
 
     /// ====================================
     /// ==== External/Public Functions =====
     /// ====================================
 
+    // @dev Initialize the Allo contract
+    //
+    // Requirements:
+    //
     function initialize(address _registry, address payable _treasury, uint256 _percentFee, uint256 _baseFee) external;
 
+    // @dev Updates the pools metadata
+    //
+    // Requirements: 'msg.sender' must be a pool admin
+    //
```

Comment:
should we add `@param` to interface functions?
we could c&p from the implementation